File: sdks/android/library/src/main/java/org/xmtp/android/library/SendOptions.kt
package org.xmtp.android.library

import org.xmtp.proto.message.contents.Content
import uniffi.xmtpv3.FfiSendMessageOpts

data class SendOptions(
    var compression: EncodedContentCompression? = null,
    var contentType: Content.ContentTypeId? = null,
    @Deprecated("This option is no longer supported and does nothing")
    var ephemeral: Boolean = false,
    // Optional idempotency key. Re-sending identical content with the same key
    // produces the same message id and is deduplicated. Defaults to a timestamp.
    var idempotencyKey: String? = null,
)

data class MessageVisibilityOptions(
    val shouldPush: Boolean,
    // Optional idempotency key. Re-sending identical content with the same key
    // produces the same message id and is deduplicated. Defaults to a timestamp.
    val idempotencyKey: String? = null,
) {
    fun toFfi(): FfiSendMessageOpts =
        FfiSendMessageOpts(shouldPush = shouldPush, idempotencyKey = idempotencyKey)
}

File: sdks/ios/Sources/XMTPiOS/SendOptions.swift
//
//  SendOptions.swift
//
//
//  Created by Pat Nakajima on 1/19/23.
//

import Foundation

public struct SendOptions {
	public var compression: EncodedContentCompression?
	public var contentType: ContentTypeID?
	public var ephemeral = false
	/// Optional idempotency key. Re-sending identical content with the same key
	/// produces the same message id and is deduplicated. Defaults to a timestamp.
	public var idempotencyKey: String?

	public init(compression: EncodedContentCompression? = nil, contentType: ContentTypeID? = nil,
	            ephemeral: Bool = false, idempotencyKey: String? = nil)
	{
		self.compression = compression
		self.contentType = contentType
		self.ephemeral = ephemeral
		self.idempotencyKey = idempotencyKey
	}
}
